nnvm/python/nnvm/compiler/build_module.py

```diff
@@ -270,6 +270,20 @@ def build(graph, target=None, shape=None, dtype="float32",
     # Apply optimization
     with target:
         graph = optimize(graph, shape, dtype, layout)
+
+    # Clear extra params without nodes.
+    arg_list = []
```

Comment:
make this a function _remove_noref_params(params, graph); 

The input name can be obtained by ```graph.symbol.list_input_names()```